src/parser/Parser.ts

```diff
@@ -407,17 +411,18 @@ export class Parser {
                         funcDeclaration.func,
                         overrideKeyword
                     );
-
+                    methodStatement.annotations = fieldAnnotations;
                     //refer to this statement as parent of the expression
                     functionStatement.func.functionStatement = methodStatement;
 
                     body.push(methodStatement);
 
                     //fields
                 } else if (this.checkAny(TokenKind.Identifier, ...AllowedProperties)) {
-                    body.push(
-                        this.classFieldDeclaration(accessModifier)
-                    );
+
+                    const fieldStatement = this.classFieldDeclaration(accessModifier);
```

Comment:
Good stuff. Also need to reset `this.pendingAnnotations` in this case.

Reply:
I don't think so.. we can only acrue them in blocks, which is delt with in the previous branch.

Reply:
Hmm no if you don't consume it then it will be used by the next declaration isn't it?
Also why not use the `consumePendingAnnotation` function.

Reply:
good calll - this reads better, like @TwitchBronBron wanted, too if I do that. thanks.

Reply:
Why don't you do the same for methods?

Reply:
same reason as classes
![image](https://user-images.githubusercontent.com/1214545/104822679-a12c7d80-5844-11eb-81b7-910775ed6ad5.png)

Reply:
now I think about it - we should have cleared it out there, too..

Reply:
pushed up a  fix for it.

Reply:
Ugh look I think this whole approach kind of works but isn't the right way to go.
The problem is that basically annotations need to be saved/restored every time a "block" or "body" is entered, instead of this adhock redundant logic.

This logic will fail as well for a field with a `sub` as initial value 

Oh and it was merged. Well... I'll need to refactor that.

Reply:
It's quite an edge case though. I think annotations broadly working on classes except for one edge case is better than totally broken annotations on classes. 

As you say, we can always fix the broken edge case in future, and if needs be adopt a different approach at that juncture. You should file a bug for the field sub case. Hadnt even dawned on me that someone would ever do that (surely a field defined as a sub is just a function, so it's the edgiest of edge cases one could contrive - I doubt we'll ever see that bug reported).